### PR TITLE
[DBAAS-974] upgrade envtest

### PR DESCRIPTION
**Pull request by tchughesiv**
Signed-off-by: Tommy Hughes <tohughes@redhat.com>

### CODE REVIEW

File: controllers/dbaas_reconciler_test.go [hunk 1]
```diff
@@ -86,9 +86,9 @@ var _ = Describe("Get DBaaSProvider", func() {
 			Provider: v1alpha1.DatabaseProvider{
 				Name: "test-provider",
 			},
-			InventoryKind:                "testInventoryKind",
-			ConnectionKind:               "testConnectionKind",
-			InstanceKind:                 "testInstanceKind",
+			InventoryKind:                "MongoDBAtlasInventory",
+			ConnectionKind:               "MongoDBAtlasConnection",
+			InstanceKind:                 "MongoDBAtlasInstance",
```
Comment (xieshenzh) [inline controllers/dbaas_reconciler_test.go:91]: What is the reason for changing these values?

File: controllers/dbaasprovider_controller_test.go [hunk 2]
```diff
@@ -31,103 +31,87 @@ import (
...
 			Version: v1alpha1.GroupVersion.Version,
-			Kind:    createdInstanceKind,
+			Kind:    "MongoDBAtlasInstance",
 		})
 		inOwner := &v1alpha1.DBaaSInstance{}
 
-		BeforeEach(func() { assertNotWatched(iSrc, iOwner, cSrc, cOwner, inSrc, inOwner) })
-		BeforeEach(assertResourceCreation(provider))
-		AfterEach(assertResourceDeletion(provider))
-		AfterEach(func() { reset(iSrc, iOwner, cSrc, cOwner, inSrc, inOwner) })
```
Comment (xieshenzh) [inline controllers/dbaasprovider_controller_test.go:83]: What is the reason for this change? The assertion, deletion and reset are intentional and necessary.

File: Makefile [hunk 3]
```diff
@@ -202,12 +219,14 @@ clean-namespace:
 	-oc delete dbaasconnections.dbaas.redhat.com --all
 	-oc delete dbaasservices.dbaas.redhat.com --all
 
-CONTROLLER_GEN = $(shell pwd)/bin/controller-gen
-controller-gen: ## Download controller-gen locally if necessary.
-	$(call go-get-tool,$(CONTROLLER_GEN),sigs.k8s.io/controller-tools/cmd/controller-gen@v0.4.1)
+.PHONY: controller-gen
+controller-gen: $(CONTROLLER_GEN) ## Download controller-gen locally if necessary.
+$(CONTROLLER_GEN): $(LOCALBIN)
+	test -s $(LOCALBIN)/controller-gen || GOBIN=$(LOCALBIN) go install sigs.k8s.io/controller-tools/cmd/controller-gen@$(CONTROLLER_TOOLS_VERSION)
```
[the remark this answers is not in the document]
Comment (tchughesiv) [reply Makefile:225]: Because only the latest v4 kustomize releases offer additional platform binaries, like darwin arm64. 
https://github.com/kubernetes-sigs/kubebuilder/issues/2566

So, for now we'll keep kustomize as-is.

File: controllers/dbaasprovider_controller_test.go (hunk 2 above)
Comment (tchughesiv) [reply controllers/dbaasprovider_controller_test.go:83]: This has to do w/ the envtest upgrade. One of the problems is that kinds like "createdInventoryKind" aren't valid CRDs. So, I've had to switch to using valid CRDs like Mongo/Crunchy.

Once this switch was done, every one of these dbaasprovider_controller tests were failing. So, I've had to tweak the unit tests accordingly.

That said, this is why I wanted your review here as you are the expert wrt this section of the code. Could you fork my dev branch and make any improvements you deem necessary? thanks

File: controllers/dbaas_reconciler_test.go (hunk 1 above)
Comment (tchughesiv) [reply controllers/dbaas_reconciler_test.go:91]: This has to do w/ the envtest upgrade. It requires valid/known CRDs in order to process these tests. So, I've had to switch to using valid CRDs like Mongo/Crunchy.

File: controllers/dbaasprovider_controller_test.go (hunk 2 above)
Comment (xieshenzh) [reply controllers/dbaasprovider_controller_test.go:83]: Please try this: https://github.com/tchughesiv/dbaas-operator/pull/8

Comment (tchughesiv) [reply controllers/dbaasprovider_controller_test.go]: its been merged, thanks

File: Makefile (hunk 3 above)
[the remark this answers is not in the document]
Comment (jianrongzhang89) [reply Makefile:225]: OK

**Review by xieshenzh — APPROVED** [review]
